### DISCUSSION TITLE
`llama-agents` Roadmap

### DISCUSSION THREAD

**Discussion by logan-markewich** [run-llama/llama_deploy#49] [Announcements]
Welcome to `llama-agents`!

With this new repository, we are laying the ground work for our vision of the agentic future -- agents running as micro-services, orchestrating together to complete tasks in a distributed network.

As part of this framework, we want to build things out in a way that best serves the needs of developers and the community. To that end, we are sharing a public roadmap and vision for the project, as well as some initial priorities for each pillar.

## 1. Robustness and Scalability
Our first pillar focuses on strengthening the core infrastructure of llama-agents to support larger and more complex deployments.

- Integrating RabbitMQ as an additional message queue option [#25]
- Improving the acknowledgment and response process in the message queue [#38]
- Implementing more efficient task processing mechanisms [#42]

These enhancements will ensure that llama-agents can handle increased load and more diverse use cases with improved reliability and performance.

## 2. Extensibility and Customization
The second pillar aims to provide developers with greater flexibility to tailor llama-agents to their specific needs.

- Creating a custom Service class for more tailored implementations [#28]
- Developing an example of a custom orchestrator [#37]

These additions will showcase the framework's adaptability and empower users to create specialized solutions within the llama-agents ecosystem.

## 3. Documentation and Accessibility
Our third pillar is dedicated to making `llama-agents` more accessible and easier to use, especially for newcomers to the framework.

- Creating comprehensive initial documentation [#30]
  - this includes examples, API reference, architecture diagrams, and other helpful content!
  - this also includes implementing latest research in agentic workflows
- Adding docstrings to all classes and methods [#39]
- Benchmarking on agentic datasets

## 4. Community Engagement and Feedback
The final pillar recognizes the importance of our user community in shaping the future of llama-agents.

- Actively soliciting and incorporating user feedback
- Regularly updating the roadmap based on community needs
- Encouraging contributions from the open-source community

We believe that engaging with our users and the broader developer community is crucial for the long-term success and relevance of `llama-agents`.

As we progress through these pillars, we'll be updating specific tickets and milestones in our GitHub repository. We encourage our community to stay engaged, provide feedback, and contribute to the ongoing development of llama-agents.

You can check the latest prioritized tickets on the [project board](https://github.com/orgs/run-llama/projects/5).

**Comment by ogahozy** [comment]
Great
